### DISCUSSION TITLE
Why do we have to use swizzle

### DISCUSSION THREAD

**Discussion by xws117** [NVIDIA/cutlass#1464] [Q&A]
In the talk of cutlass,( https://developer.nvidia.com/gtc/2020/video/s21745)  ,for load data from Gmem to smem, to avoid bank conflict ,use xor swizzle to put data in different position.  But the whole porcession need 4 phases to finish , if we do not use the swizzle ,there will be 4way-bank conflict, also take 4 clock period,so what the advantage of swizzle ,thanks for you answer

**Comment by thakkarV** [comment]
A bank conflict does not simply cost 1 extra cycle, it is more than that. While not strictly multiplicative, accesses are serialized in cases of bank conflicts
